src/mail/view/MailViewer.ts

```diff
@@ -387,17 +398,37 @@ export class MailViewer implements Component<MailViewerAttrs> {
 					"line-height": this.bodyLineHeight ? this.bodyLineHeight.toString() : size.line_height,
 					"transform-origin": "top left",
 				},
-			},
-			m.trust(sanitizedMailBody),
+			}
 		)
 	}
 
+	/**
+	 * We render the sanitized mail body to a shadow dom to ensure that any styles don't leak into the rest of the app.
+	 * Styles in mailbodies *should* get sanitized out, but even with sanitization, there is always the possibility of bugs,
+	 * either our own or in the sanitization, so we do this as a safety precaution
+	 */
+	private renderToShadowRoot(sanitizedMailBody: string) {
```

Comment:
you switched back to stringifying but could we do `DocumentFragment.cloneNode(true)` instead?

Reply:
considered it, is it ok to do it every time?

Reply:
Well it is cheaper than parsing it each time?

Reply:
true